ui/src/main/webapp/wizards/ldap/stages/network-settings.js

```diff
@@ -32,87 +43,163 @@ const testConnect = (conn) => ({
+          where the first host is considered the primary host.  The configuration of each host
+          in the cluster is assumed to be the same (i.e. bind user, directory settings, etc).
+        </Description>
+
+        <Body>
+          <Select
+            value={configs.encryption}
+            errorText={errors.encryption}
+            onEdit={onEdit('encryption')}
+            label='Encryption Method'
+            options={[ 'none', 'ldaps', 'startTls' ]} />
+
+          <Hostname
+            value={configs.hostname}
+            errorText={errors.hostname}
+            onEdit={onEdit('hostname')}
+            autoFocus />
+
+          <Port
+            value={configs.port}
+            errorText={isPortInvalid ? getFriendlyMessage('INVALID_PORT_RANGE') : errors.port}
+            onEdit={onEdit('port')}
+            options={[389, 636]} />
+
+          <Select
+            value={configs.loadbalancing}
+            onEdit={onEdit('loadbalancing')}
+            label='Load Balancing Algorithm'
+            options={[ 'roundRobin', 'failover' ]} />
+
+          {messages.map((msg, i) => <Message key={i} {...msg} />)}
+
+          <RaisedButton
+            primary
+            style={{margin: '0 auto', marginBottom: '30px', marginTop: '10px', display: 'block'}}
+            label='Add Host'
+            disabled={isPortInvalid}
             onClick={() => {
               onStartSubmit()
               client.query(testConnect({
                 hostname: configs.hostname,
                 port: configs.port,
                 encryption: configs.encryption
-              }))
-                .then(() => {
-                  onEndSubmit()
-                  next('bind-settings')
-                })
-                .catch((err) => {
-                  onEndSubmit()
-                  onError(err)
+              })).then((result) => {
+                connectionInfo.push([configs.hostname, configs.port])
+                onEdit({
+                  hostname: '',
+                  messages: [],
+                  connectionInfo: connectionInfo
                 })
-            }}
-          />
-        </Navigation>
-        {messages.map((msg, i) => <Message key={i} {...msg} />)}
-      </Body>
-    </div>
-  )
+                onError([])
+                onEndSubmit()
+              }).catch((err) => {
+                onError(err)
+                onEndSubmit()
+              })
+            }} />
+
+          <Table multiSelectable onRowSelection={(selected) => this.setState({ selected })}>
+            <TableHeader displaySelectAll={false} >
+              <TableRow>
+                <TableHeaderColumn>Host</TableHeaderColumn>
+                <TableHeaderColumn>Port</TableHeaderColumn>
+              </TableRow>
+            </TableHeader>
+            <TableBody showRowHover deselectOnClickaway={false}>
+              {Object.keys(connectionInfo).map((connInfo, i) =>
```

Comment:
Since `connectionInfo` is already a list, we don't need to get the keys to map over it.